src/checks/cargo/cargo.ts

```diff
@@ -50,40 +94,29 @@ export function createAnnotationFromManifestError(item: CargoManifestParseError)
   }
 }
 
-export function getCompilerAnnotations(item: CargoCompilerMessage): CheckAnnotation[] {
-  // The `children` i.e. the child diagnostic messages can safely
-  // be ignored, as they only provide additional information to
-  // the main diagnostic. Such as "machine information" which can
-  // be used to automatically resolve the issue.
-  const diagnostic = item.message
-  const span = getPrimaryOrFirstSpan(diagnostic.spans)
-
-  // If the diagnostic has no spans, then it implies that
-  // the issue exists outside of this crate. Such as a
-  // warning in a dependency. If the dependency is one of
-  // our own crates, then this issue will already have been
-  // caught.
-  if (span == null) {
-    return []
-  }
-
-  const annotations: CheckAnnotation[] = []
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export function filterNone<T>(item: T): boolean {
+  return true
+}
 
-  annotations.push({
-    path: span?.file_name ?? 'Cargo.toml',
-    start_line: span?.line_start ?? 0,
-    end_line: span?.line_end ?? 0,
-    annotation_level: 'failure',
-    message: diagnostic.rendered ?? diagnostic.message,
-    title: diagnostic.message,
-    raw_details: JSON.stringify(diagnostic, null, 4)
-  })
+export function filterClippyLints(item: CargoCompilerMessage): boolean {
+  if (item.message.rendered?.includes('clippy')) {
+    return false
+  } else {
+    return true
+  }
+}
 
-  return annotations
+export function filterNonClippyLints(item: CargoCompilerMessage): boolean {
+  // All clippy lints contains "clippy" in their message, so
+  // filter all compiler messages that does not contain
+  // "clippy". This is to avoid repeating all issues already
+  // found by `cargo check`.
+  return !filterClippyLints(item)
 }
```

Comment:
I'm not sure why do you need the negation as well? As far as i can see, it is only used once, and it is negated as well

Reply:
I think we should keep this, maybe it should be renamed to `filterCargoBuildLints`. But just because the implementation uses `filterClippyLints`, that doesn't mean the implementation can't change. So I think we should keep both, but again, I can rename it, if you prefer the other name

Reply:
It's not needed, the name is suggestive enough